patches/server/0969-New-event-verbose-logging-command.patch

```diff
@@ -0,0 +1,204 @@
+new file mode 100644
+index 0000000000000000000000000000000000000000..165bb5324aa06e4bd204a570b45b185b0bd11a84
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/command/subcommands/EventVerboseCommand.java
+@@ -0,0 +1,160 @@
++package io.papermc.paper.command.subcommands;
++
++import com.destroystokyo.paper.event.server.ServerTickEndEvent;
++import com.destroystokyo.paper.event.server.ServerTickStartEvent;
++import com.google.common.collect.Lists;
++import com.google.gson.JsonArray;
++import com.google.gson.JsonObject;
++import com.google.gson.internal.Streams;
++import com.google.gson.stream.JsonWriter;
++import io.papermc.paper.command.PaperSubcommand;
++import net.minecraft.server.MinecraftServer;
++import org.bukkit.command.CommandSender;
++import org.bukkit.event.Event;
++import org.bukkit.inventory.ItemStack;
++import org.checkerframework.checker.nullness.qual.NonNull;
++import org.checkerframework.checker.nullness.qual.Nullable;
++import org.checkerframework.framework.qual.DefaultQualifier;
++
++import java.io.IOException;
++import java.io.PrintStream;
++import java.io.StringWriter;
++import java.lang.reflect.Field;
++import java.lang.reflect.Modifier;
++import java.nio.charset.StandardCharsets;
++import java.nio.file.Files;
++import java.nio.file.Path;
++import java.time.LocalDateTime;
++import java.time.format.DateTimeFormatter;
++import java.util.Collection;
++import java.util.LinkedList;
++import java.util.List;
++import java.util.Map;
++import java.util.stream.Stream;
++
++import static net.kyori.adventure.text.Component.text;
++import static net.kyori.adventure.text.format.NamedTextColor.GREEN;
++import static net.kyori.adventure.text.format.NamedTextColor.RED;
++
++@DefaultQualifier(NonNull.class)
++public class EventVerboseCommand implements PaperSubcommand {
++
++    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH.mm.ss");
++    @Nullable private Path currentFile;
++    //have to be static to static call from paperpluginmanagerimpl
++    private static boolean isRunning = false;
++    private static final LinkedList<Wrapper> toDump = Lists.newLinkedList();
++    //ignore unnecessary events
++    private static final List<Class<? extends Event>> exclude = List.of(ServerTickStartEvent.class,
++        ServerTickEndEvent.class);
++
++
++    @Override
++    public boolean execute(CommandSender sender, String subCommand, String[] args) {
++        if(!isRunning) {
++            isRunning = true;
++            Path parent = Path.of("debug");
++            Path path = parent.resolve("event-verbose-" + FORMATTER.format(LocalDateTime.now()) + ".txt");
++            try {
++                Files.createDirectories(parent);
++                currentFile = Files.createFile(path);
++                sender.sendMessage(text("Now storing event verbose, run this command again to save your file", GREEN));
```

Comment:
`Now storing event verbose` doesn't make sense. Probably best to be more descriptive as to what it's actually doing.